src/protocol/damage.rs

```diff
@@ -92,12 +82,6 @@ impl From<ReportLevel> for Option<u32> {
         Some(u32::from(input.0))
     }
 }
-impl From<bool> for ReportLevel {
-    #[inline]
-    fn from(value: bool) -> Self {
-        Self(value.into())
-    }
-}
 impl From<u8> for ReportLevel {
     #[inline]
     fn from(value: u8) -> Self {
```

Comment:
I can't comment on something a few lines outside, so I leave this comment here:
```rust
impl TryFrom<u16> for ReportLevel {
    type Error = ParseError;
    fn try_from(value: u16) -> Result<Self, Self::Error> {
        u8::try_from(value).or(Err(ParseError::InvalidValue)).map(Self)
    }
}
```
Now that we do not really check the value any more, is `ConversionFailed` more fitting than `InvalidValue`? The docs for the two are:
```
    /// A value did not fit.
    ///
    /// This error can e.g. happen when a value that was received from the X11 server does not fit
    /// into an `usize`.
    ConversionFailed,

    /// A value was outside of its valid range.
    ///
    /// When parsing the value of an enumeration, not all possible integer values have a defined
    /// meaning. This error occurs when an invalid value is encountered in this context.
    ///
    /// For example, `xproto` has an enumeration `Place` with possible values `OnTop` (0) and
    /// `OnBottom` (1). Any value other than 0 or 1 generates an `InvalidValue` when parsing.
    InvalidValue,
```
(Slightly unrelated, but I think the docs for `InvalidValue` need an overhaul. I am sure that there are still places that use it (`<switch>` parsing?), but the example in the docs is now wrong)